### DISCUSSION TITLE
Why there is no command to (P)reset Sonoff SPM Main EnergyToday?

### DISCUSSION THREAD

**Discussion by ADEXITUM** [arendst/Tasmota#22287] [Support request] [answered]
I want  to reset or fetch today energy on Tasmota 14.2.0 (release-tasmota32)

Why there's commands SspmEnergyTotal and SspmEnergyYesterday, but no SspmEnergyToday?

On the web interface EnergyToday, EnergyTotal and EnergyYesterday are calculated separately, how to obtain or change EnergyToday using http command?

Is there any workaround?

**Comment by sfromis** [comment]
The general way of getting sensor data, including energy monitoring data, is the command `Status 10` - available same places as other commands.

**Reply by ADEXITUM** [reply]
Thank you! Could you also answer: 
1. Is there any way to reset all energy monitoring data for specific relay? As I see in documentation for Sonoff SPM Main there's only `SspmEnergyTotal` command which doesn't affect today energy, so even after reset `SspmEnergyTotal` is calculated of 0 + today energy. I'd like to zero all of these values. Commands `EnergyToday`, `EnergyYesterday`, `EnergyTotal` are unknown.
2. Is there command to get monitoring data for some period, e.g. from `2024.10.17 22:00` to` 2024.10.18 04:00`? 

I couldn't find these in documantation.

**Reply by sfromis** [reply]
As Tasmota does not save history of energy consumption, outside of total/yesterday, no way of querying a specific interval, unless you create such yourself, or base it on already published MQTT telemetry data.

As I'm not using the Sonoff SPM, my previous answer was very deliberately phrased as "general", not trying to be a direct answer about what you can or can not get on that particular device, which is outside of the broader Tasmota support for energy monitoring.

**Reply by ADEXITUM** [reply]
Could you please provide some information on possible ways of resetting all power monitoring data including EnergyToday?
And was it a hardware limitation of SPM Main which made it impossible to reset today energy? Just want to save time in case I'll decide to try implement such thing myself. Maybe someone who developed functionality related to SPM Main may help?

Are you going to add such functionality in future releases?

**Reply by arendst** [reply, marked as answer]
There is no way to reset EnergyToday as the ARM devices always provide EnergyToday value.

I'm not going to add this one-time only activity; tomorrow there will be a new energy today.

In addition, the ARM's keep counting their energy anyway so there really is no need to reset today's energy.

**Reply by ADEXITUM** [reply]
Thank you for your reply. I've got one question left: SPM Main tracks power consumption only for the period up to 180 days. 
Does Tasmota SspmTotalEnergy acts the same or the value is updated infinetily?

**Comment by arendst** [comment]
The latter until float rolls over
